sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -454,6 +490,40 @@ case class BroadcastHashJoinExec(
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
 
+    if (isNullAwareAntiJoin) {
+      if (broadcastRelation.value.isInstanceOf[EmptyHashedRelation]) {
+        return s"""
+                  |// If the right side is empty, NAAJ simply returns the left side.
+                  |$numOutput.add(1);
+                  |${consume(ctx, input)}
+            """.stripMargin
+      } else if (broadcastRelation.value.isInstanceOf[EmptyHashedRelationWithAllNullKeys]) {
+        return s"""
+                  |// If the right side contains any all-null key, NAAJ simply returns Nil.
```

Comment:
nit: Is it idiomatic to use the word `Nil` instead of Nothing ?

Reply:
will do the update